### PR TITLE
chore: Add style enforcer

### PR DESCRIPTION
**Pull request by danjohnson95**
There were lots of styling errors in [my other PR](https://github.com/UserScape/larajobs-desktop/pull/13), sorry!

Here's a style enforcer that'll run when a PR is raised.

It's using PHP_CodeSniffer, and the rules I've specified are:

- PSR1
- PSR2
- PSR12

I've included these since they're mentioned as requirements in [Spatie's docs](https://spatie.be/guidelines/laravel-php), as referenced in the [Contributing](https://github.com/UserScape/larajobs-desktop#contributing) section of the README file in this repo.

### CODE REVIEW

**Comment by danjohnson95** [comment]
Build is currently failing due to failing tests, this will pass once https://github.com/UserScape/larajobs-desktop/pull/15 is merged in

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        uses: actions/checkout@v3
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
+          tools: cs2pr, phpcs
+
+      - name: Run phpcs
+        run: phpcs -q --report=checkstyle app | cs2pr
```
Comment (LukeTowers) [inline .github/workflows/style.yml]:
```suggestion
        run: phpcs -q --report=checkstyle app | cs2pr

```

**Comment by LukeTowers** [comment]
Thanks very much @danjohnson95! 

Do you think you'd be able to figure out how to add a workflow for creating a build and attaching the built packages to the release when new releases are published? 

Ideally I'd like the release process to be we create a new release through the github UI which creates a tag and then triggers the workflow to run the necessary build / publish commands and copies the relevant files from the /dist directory to the release as downloadable files for people to install on their computers without having to build from the source.

@mpociot side note, do you know if it's possible to get the NativePHP updater to source from GitHub releases instead of DO Spaces / AWS? That would be absolutely incredible and a game changer if we could get that working, especially if we can also provide an example github workflow file for open source NativePHP applications to reference to truly make the distribution process of NativePHP applications as easy as publishing releases on GitHub.

**Comment by mpociot** [comment]
Yes! That's definitely possible, I just need to add it 😅

**Comment by LukeTowers** [comment]
@mpociot consider this my pretty pretty please 😂 ❤️

**Comment by danjohnson95** [comment]
Yeah I can do that! I was actually having a play with compilation last night - I noticed in the README you mentioned that the `MenuBar` wasn't working when the application was compiled, but I couldn't replicate that bug. Seemed to be working fine for me when compiled.

We'll need an Apple Developer certificate (requires an active Apple Developer subscription) added to the Secrets in the settings of this repo.

**Comment by LukeTowers** [comment]
@danjohnson95 yeah, @mpociot already fixed that issue regarding the menubar not showing up, the build tool was missing a file that hadn't been checked in yet. 

Do you have a link to the precise instructions required for setting up the secret in the repo so that @ianlandsman can take care of that? I don't have the necessary permissions.

**Comment by ianlandsman** [comment]
Yeah @danjohnson95 @LukeTowers UserScape is already in the apple developer program, but I've never added a secret for it to Github. If you can send me the instructions link I'll get it setup.

**Comment by LukeTowers** [comment]
@ianlandsman this is probably the most relevant link on the topic, hopefully the instructions are straightforward enough: https://github.com/marketplace/actions/electron-builder-action#code-signing

**Comment by ianlandsman** [comment]
Tried this out this morning, but none of the instructions work. There's nowhere to set a password on the generated certificate. There are also many types of certificates but the instructions don't say which to use. Will try and dig in deeper later this week, but for now won't be able to add the certs.

**Comment by danjohnson95** [comment]
@ianlandsman I've written a blog post explaining how to generate the required certificates - https://danjohnson.xyz/en/blog/signing-your-native-php-application

**Comment by LukeTowers** [comment]
@danjohnson95 that's fantastic, great job! That article will be essential to the NativePHP developer community :) @mpociot do you think you could share it?

**Comment by danjohnson95** [comment]
FYI there are some issues blocking the ability to package in a CI environment. I've [raised a bug](https://github.com/NativePHP/laravel/issues/179) in the base repo and I'm working on fixes.

**Comment by ianlandsman** [comment]
@danjohnson95 great write up! I tried doing it but get an untrusted cert warning after importing back into my computer every time. Really bizarre. It doesn't seem like there are any steps that could really be messed up in there. Any thoughts? I did the entire process 4+ times but no dice. 
Screenshot on 2023-08-15 at 15-37-56

**Comment by danjohnson95** [comment]
Hmm I recognise that error, I think I had something similar when dealing with PassKit. From memory, it was because I was missing one of Apple's Intermediate certificates. You can download them here - https://www.apple.com/certificateauthority/ _Preeety_ sure it's the "Worldwide Developer Relations" ones we're interested in.

Could you try downloading and importing those? Let me know if that works and I'll mention it in my blog post.

**Comment by ianlandsman** [comment]
@danjohnson95 OK got it setup and the secrets in place as per your article. I needed one of the intermediate certs. I grabbed a few not sure which is was. Did the worldwide cert but also the Developer ID and Developer Authentication.

Also, to actually export a .p12 I had to go to "My Certificates", toggle it open, and select my specific key and then I could export from there as .p12. Otherwise it was greyed out. Pretty much as described here https://fluffy.es/keychain-cant-export-to-p12/

**Comment by danjohnson95** [comment]
Good to know, thanks! I'll get that included in my article.

There's one more set of credentials we'll need (I've just added instructions to my article). The certificate you've added is for _codesigning_, but we'll also need another set for _notarising_.

In order to run an application on macOS 10.15 or higher, Apple requires it to be notarised. To notarise our application in CI, we need to add three more secrets

Full instructions available [in the article](https://danjohnson.xyz/en/blog/signing-your-native-php-application), but TLDR:

- `NATIVEPHP_APPLE_ID` - Your Apple ID (email) associated with your Apple Developer account
- `NATIVEPHP_APPLE_ID_PASS` - An "App-specific password" generated on Apple's website
- `NATIVEPHP_APPLE_TEAM_ID` - Your Apple Developer Team ID

Once those are in and my PR addressing some bugs in the Electron package (https://github.com/NativePHP/electron/pull/74) are merged in, we're all set for compiling the application in GitHub Actions for macOS!

Here's a sneak peak of the compiled application uploaded as an asset using my Apple ID credentials to sign and notarise it, and using my fork of the Electron package

https://github.com/danjohnson95/larajobs-desktop/actions/runs/5883872020

**Comment by danjohnson95** [comment]
Also, now that NativePHP supports Windows, do we want this application to support Windows too? Codesigning for Windows applications is a lot more complicated, but I've got experience in it and would love to help out the community here. Happy to move this to a Discussion or an Issue so we can better keep track of this chat.

**Comment by ianlandsman** [comment]
@danjohnson95 did an earlier version of the article not have those other elements? Swear those weren't there 🤣

OK those are added.

I'm a no on Windows for now.

**Comment by LukeTowers** [comment]
@ianlandsman Personally I'd like to see the codesigning and deployment artifacts being used for Windows as well, this repo could be a good "real-world" demo of NativePHP and the processes for actually taking a NativePHP application to the finish line of cross platform distribution. 

While it may not matter specifically to LaraJobs about having a Windows release; it does provide motivation for the open source contributors to add that functionality so that there's a reference to point to on how a real-world application actually handles those steps. I know I personally would be interested in seeing that @danjohnson95; although I don't use Windows for development I know a LOT of developers, especially those getting started, that do because it simply is a much cheaper and easier platform to get a hold of when you're first starting out or when you're in regions of the world with less economic power.

**Comment by ianlandsman** [comment]
@LukeTowers @danjohnson95 OK OK 😄 No way I'll have time to figure this out, but if Dan is up for providing some insanely detailed instructions I'm down with it.

**Comment by LukeTowers** [comment]
@danjohnson95 side note, tried the built application and ran into the following error when launching: https://flareapp.io/share/17WN4zWP

**Comment by danjohnson95** [comment]
Haha no worries, more than happy to provide some insanely detailed instructions. 

However word of warning - the certificate required to sign applications for Windows are super expensive. Plus, new legislation means that the certificates need to be issued on either a hardware token or through the CA's key server. In order to do automated builds in CI, we'll need the certificate hosted on their key server. So looking at around $600 USD per year https://order.digicert.com/step1/code_signing

**Comment by LukeTowers** [comment]
@danjohnson95 is code signing required for Windows applications?

**Comment by danjohnson95** [comment]
Not a requirement. If the user opens an application that hasn't been code signed, Windows will just give lots of warnings that the application is unsafe and they'll have to find the tucked away "Run anyway" button. Anti-virus software will also do lots of complaining.

Regarding the error, that's a weird one. I'll look into that now 👀

**Comment by ianlandsman** [comment]
OK well.... 

I'm not sure about $600/year. Perhaps we start out doing what is necessary to build the windows version with warnings. I don't think it's a HUGE issue since the code signing doesn't actually add any benefits to the idea of this being a solid NativePHP example app. 

Then if it grows beyond being an example app on windows and there's demand to remove the warnings we could do the code signing.

**Pull request merged**